process/block/metablock.go

```diff
@@ -2210,10 +2210,19 @@ func (mp *metaProcessor) computeExistingAndRequestMissingShardHeaders(metaBlock
 			mp.hdrsForCurrBlock.highestHdrNonce[shardData.ShardID] = hdr.GetNonce()
 		}
 
+		shouldConsiderProofsForNotarization := mp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, hdr.GetEpoch())
+		hasProofForShardHeader := mp.proofsPool.HasProof(shardData.ShardID, shardData.HeaderHash)
+		if shouldConsiderProofsForNotarization && !hasProofForShardHeader {
+			// if there is no proof for current shard header, request the next one that holds this proof
+			mp.hdrsForCurrBlock.missingFinalityAttestingHdrs++
+			go mp.requestHandler.RequestShardHeaderByNonce(hdr.GetShardID(), hdr.GetNonce()+1)
+		}
+
 		mp.updateLastNotarizedBlockForShard(hdr, shardData.HeaderHash)
 	}
 
-	if mp.hdrsForCurrBlock.missingHdrs == 0 {
+	requestedFinalityAttestingBasedOnProofs := mp.hdrsForCurrBlock.missingFinalityAttestingHdrs > 0
```

Comment:
I am not sure that this has no side effect on behavior before activation

this is mostly because it would change the same `missingFinalityAttestingHdrs` variable which gets updated also with legacy treatment

Reply:
indeed, updated